src/source.c

```diff
@@ -1881,17 +1918,12 @@ _dispatch_timers_program2(uint64_t nows[], _dispatch_kevent_qos_s *ke,
 		_dispatch_trace_next_timer_set(
 				TAILQ_FIRST(&_dispatch_kevent_timer[tidx].dk_sources), qos);
 		_dispatch_trace_next_timer_program(delay, qos);
-		delay += _dispatch_source_timer_now(nows, DISPATCH_TIMER_KIND_WALL);
-		if (slowpath(_dispatch_timers_force_max_leeway)) {
-			ke->data = (int64_t)(delay + leeway);
-			ke->ext[1] = 0;
-		} else {
-			ke->data = (int64_t)delay;
-			ke->ext[1] = leeway;
-		}
-		ke->flags |= EV_ADD|EV_ENABLE;
-		ke->flags &= ~EV_DELETE;
+
+                kevent_set_delay(ke,delay,leeway,nows);
 	}
+ 
+	ke->flags |= EV_ADD|EV_ENABLE;
```

Comment:
bogus, should stay in the previous scope. I would even argue it should be done by the new static inline you just added